chrome/content/zotero/zoteroPane.js

```diff
@@ -4151,74 +4153,277 @@ var ZoteroPane = new function()
+			addToCollectionMenuItem.label = Zotero.getString("zotero.toolbar.newCollection.label");
+			addToCollectionMenuItem.addEventListener("command", (_) => {
+				ZoteroPane.addSelectedItemsToCollection(library, true);
+			});
+			// Set value to be able to find this menuitem if we need to disable it
+			addToCollectionMenuItem.value = `new-collection-L${library.libraryID}`;
+			// Insert menuitem in specified location
+			if (insertAfter) {
+				insertAfter.after(addToCollectionMenuItem);
+			}
+			else {
+				parent.prepend(addToCollectionMenuItem);
+			}
+			// Surround it with separators
+			if (addToCollectionMenuItem.previousElementSibling
+				&& addToCollectionMenuItem.previousElementSibling.tagName !== "menuseparator") {
+				let topSeparator = document.createXULElement("menuseparator");
+				addToCollectionMenuItem.previousElementSibling.after(topSeparator);
+			}
+			if (addToCollectionMenuItem.nextElementSibling
+				&& addToCollectionMenuItem.nextElementSibling.tagName !== "menuseparator") {
+				let bottomSeparator = document.createXULElement("menuseparator");
+				addToCollectionMenuItem.after(bottomSeparator);
+			}
+		};
 
 		let items = Zotero.Items.keepTopLevel(this.getSelectedItems());
 		let collections = Zotero.Collections.getByLibrary(this.getSelectedLibraryID());
 		for (let col of collections) {
 			let menuItem = Zotero.Utilities.Internal.createMenuForTarget(
 				col,
 				popup,
-				null,
-				(event, collection) => {
-					if (event.target.tagName == 'menuitem') {
-						this.addSelectedItemsToCollection(collection);
-						event.stopPropagation();
-					}
-				},
-				collection => items.every(item => collection.hasItem(item))
+				{
+					clickAction: (event, collection) => {
+						if (event.target.tagName == 'menuitem') {
+							this.addSelectedItemsToCollection(collection);
+							event.stopPropagation();
+						}
+					},
+					disabledCheck: collection => items.every(item => collection.hasItem(item))
+				}
 			);
 			popup.append(menuItem);
 		}
 
-		separator.setAttribute('hidden', !collections.length);
-	};
+		// New Collection
+		addNewCollectionMenuItem(popup, null, Zotero.Libraries.get(Zotero.Libraries.userLibraryID));
+
+		// My Publications
+		let myPublicationMenuItem;
+		if (this.getSelectedLibraryID() == Zotero.Libraries.userLibraryID) {
+			myPublicationMenuItem = document.createXULElement("menuitem");
+			myPublicationMenuItem.setAttribute("label", Zotero.getString("pane.collections.publications"));
+			myPublicationMenuItem.addEventListener("command", (_) => {
+				this.addSelectedItemsToMyPublications();
+			});
+			myPublicationMenuItem.setAttribute("image", "chrome://zotero/skin/16/universal/publications.svg");
+			popup.append(myPublicationMenuItem);
+			// Cannot add top-level notes/attachment or items that are already in my publications
+			if (items.some(item => item.isNote() || item.isAttachment() || item.inPublications)) {
```

Comment:
Better to do `!item.isRegularItem()`, so this isn't accidentally enabled for some other item type later.